sympy/core/expr.py

```diff
@@ -3015,6 +3015,8 @@ def series(self, x=None, x0=0, n=6, dir="+", logx=None, cdir=0):
                 o = s1.getO()
                 s1 = s1.removeO()
             else:
+                if s1.has(Order):
+                    return s1
```

Comment:
```suggestion
            elif s1.has(Order):
                # asymptotic expansion
                return s1
            else:
```
To bring forth the three cases:
* tailing order term
* embedded order term
* no order term